source/extensions/filters/http/cache/http_cache.h

```diff
@@ -229,8 +229,17 @@ class InsertContext {
                           bool end_stream) PURE;
 
   // Inserts trailers into the cache.
-  virtual void insertTrailers(const Http::HeaderMap& trailers) PURE;
-
+  virtual void insertTrailers(const Http::ResponseTrailerMap& trailers) PURE;
+
+  /**
+   * This routine is called prior to an InsertContext being destroyed. InsertContext is responsible
+   * for making sure that any async events are cleaned up in the context of this routine. This
+   * includes timers, network calls, etc. The reason there is an onDestroy() method vs. doing this
+   * type of cleanup in the destructor is due to the deferred deletion model that Envoy uses to
+   * avoid stack unwind complications. InsertContext must not invoke any callbacks after having
+   * onDestroy() invoked.
+   */
+  virtual void onDestroy() PURE;
```

Comment:
You removed onDestroy() in the lookup context. Did you mean to also remove it here in the InsertContext? I don't see it called?

/wait-any

Reply:
Indeed; removed.